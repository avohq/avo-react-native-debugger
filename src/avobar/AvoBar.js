import React from 'react';
import { View, Text, Image } from 'react-native';
import PropTypes from 'prop-types';
import { images } from '../images';
import { colors } from '../colors';
import {styles} from './AvoBarStyles';
import {baseStyles} from '../BaseStyles';

export default function AvoBar(props) {

    let backgroundColor = colors.background;
    if (props.hasErrors) {
        backgroundColor = colors.error;
    }

    let leftIcon = images.tick;
    if (props.hasErrors) {
        leftIcon = images.white_warning;
    }

    let rightIcon = images.drag_handle_dark;
    if (props.hasErrors) {
        rightIcon = images.drag_handle_light;
    }

    let timestampColor = colors.foregroundLight;
    if (props.hasErrors) {
        timestampColor = colors.foregroundLighter;
    }

    let eventNameColor = colors.foreground;
    if (props.hasErrors) {
        eventNameColor = colors.background;
    }

    const eventDate = new Date();
    eventDate.setTime(props.lastItemTimestamp);
    const hours = eventDate.getHours();
    const minutes = eventDate.getMinutes();
    const seconds = eventDate.getSeconds();
    const ms = eventDate.getMilliseconds();
    eventTimeString = hours + ":" + minutes + ":" + seconds + '.' + ms;

    return (
        <View style={[styles.container, {backgroundColor: backgroundColor}]}>
            <View style={baseStyles.divider} />

            <View style={styles.rowStyle}>
                <Image style={styles.icon} source={leftIcon} />
                <View style={styles.timestampAndNameContainer}>
                    <Text style={[styles.timestamp, {color: timestampColor}]}>{eventTimeString}</Text>
                    <Text style={[styles.eventname, {color: eventNameColor}]}>{props.lastItemName}</Text>
                </View>

                <Image style={styles.icon} source={rightIcon} />
            </View>

            <View style={baseStyles.divider} />
        </View>
    );
}

AvoBar.propTypes = {
    hasErrors: PropTypes.bool,
    lastItemName: PropTypes.string,
    lastItemTimestamp: PropTypes.number,
}