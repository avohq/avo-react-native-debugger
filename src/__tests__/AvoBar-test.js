import React from "react";
import renderer from "react-test-renderer";

import { testEventItem } from "./testUtils";
import AvoBar from "../avobar/AvoBar";

describe("AvoBar", () => {
  test("Renders correctly", () => {
    const tree = renderer
      .create(
        <AvoBar
          hasErrors={false}
          lastItemName={testEventItem.name}
          lastItemTimestamp={testEventItem.timestamp}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  test("Renders correctly with error", () => {
    const tree = renderer
      .create(
        <AvoBar
          hasErrors={true}
          lastItemName={testEventItem.name}
          lastItemTimestamp={testEventItem.timestamp}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  test("Renders correctly without props set", () => {
    const tree = renderer.create(<AvoBar />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
