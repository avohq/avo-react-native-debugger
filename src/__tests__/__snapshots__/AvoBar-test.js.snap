// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvoBar Renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
      },
      Object {
        "backgroundColor": "#FFFFFF",
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#E3E3E3",
        "height": 1,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "height": 52,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/img/tick.png",
        }
      }
      style={
        Object {
          "height": 16,
          "margin": 16,
          "resizeMode": "contain",
          "width": 16,
        }
      }
    />
    <View
      style={
        Object {
          "flex": 1,
          "marginLeft": 8,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 8,
              "fontWeight": "bold",
              "textAlignVertical": "center",
            },
            Object {
              "color": "#666666",
            },
          ]
        }
      >
        0:0:0.0
      </Text>
      <Text
        style={
          Array [
            Object {
              "fontSize": 16,
              "fontWeight": "bold",
              "textAlignVertical": "center",
            },
            Object {
              "color": "#000000",
            },
          ]
        }
      >
        test-name
      </Text>
    </View>
    <Image
      source={
        Object {
          "testUri": "../../../src/img/drag_handle_dark.png",
        }
      }
      style={
        Object {
          "height": 16,
          "margin": 16,
          "resizeMode": "contain",
          "width": 16,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#E3E3E3",
        "height": 1,
      }
    }
  />
</View>
`;

exports[`AvoBar Renders correctly with error 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
      },
      Object {
        "backgroundColor": "#D94553",
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#E3E3E3",
        "height": 1,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "height": 52,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/img/warning.png",
        }
      }
      style={
        Object {
          "height": 16,
          "margin": 16,
          "resizeMode": "contain",
          "width": 16,
        }
      }
    />
    <View
      style={
        Object {
          "flex": 1,
          "marginLeft": 8,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 8,
              "fontWeight": "bold",
              "textAlignVertical": "center",
            },
            Object {
              "color": "#DDDDDD",
            },
          ]
        }
      >
        0:0:0.0
      </Text>
      <Text
        style={
          Array [
            Object {
              "fontSize": 16,
              "fontWeight": "bold",
              "textAlignVertical": "center",
            },
            Object {
              "color": "#FFFFFF",
            },
          ]
        }
      >
        test-name
      </Text>
    </View>
    <Image
      source={
        Object {
          "testUri": "../../../src/img/drag_handle_light.png",
        }
      }
      style={
        Object {
          "height": 16,
          "margin": 16,
          "resizeMode": "contain",
          "width": 16,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#E3E3E3",
        "height": 1,
      }
    }
  />
</View>
`;

exports[`AvoBar Renders correctly without props set 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
      },
      Object {
        "backgroundColor": "#FFFFFF",
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#E3E3E3",
        "height": 1,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "height": 52,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/img/tick.png",
        }
      }
      style={
        Object {
          "height": 16,
          "margin": 16,
          "resizeMode": "contain",
          "width": 16,
        }
      }
    />
    <View
      style={
        Object {
          "flex": 1,
          "marginLeft": 8,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 8,
              "fontWeight": "bold",
              "textAlignVertical": "center",
            },
            Object {
              "color": "#666666",
            },
          ]
        }
      >
        NaN:NaN:NaN.NaN
      </Text>
      <Text
        style={
          Array [
            Object {
              "fontSize": 16,
              "fontWeight": "bold",
              "textAlignVertical": "center",
            },
            Object {
              "color": "#000000",
            },
          ]
        }
      />
    </View>
    <Image
      source={
        Object {
          "testUri": "../../../src/img/drag_handle_dark.png",
        }
      }
      style={
        Object {
          "height": 16,
          "margin": 16,
          "resizeMode": "contain",
          "width": 16,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#E3E3E3",
        "height": 1,
      }
    }
  />
</View>
`;
